src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -19,35 +20,84 @@
 
   public IntakeSubsystem() {
-    intakeMotorControllerOne = new MotorController("Intake Motor One", Constants.intakeMotorOneID);
-    intakeMotorControllerTwo = new MotorController("Intake Motor Two", Constants.intakeMotorTwoID); 
+    intakeMotorControllerOne = new MotorController("Intake Motor One", Constants.intakeMotorOneID, 40);
+    intakeMotorControllerTwo = new MotorController("Intake Motor Two", Constants.intakeMotorTwoID, 40);
+    //CDSWheelControllerOne = new MotorController("Wheel Motor Controller 1", Constants.intakeWheelOneID, 40);
+    //CDSWheelControllerTwo = new MotorController("Wheel Motor Controller 2", Constants.intakeWheelTwoID, 40);
+
+    initialBallSensor = new DigitalInput(Constants.initialBallSensorChannel);
+    middleBallSensor = new DigitalInput(Constants.middleBallSensorChannel);
+    finalBallSensor = new DigitalInput(Constants.finalBallSensorChannel); 
 
-    intakeMotorControllerTwo.getSparkMax().follow(intakeMotorControllerTwo.getSparkMax());
+    DigitalInput[] sensorArray = {initialBallSensor, middleBallSensor, finalBallSensor};
+
+    // Remove invert=true parameter if wheels aren't running correctly
+    //CDSWheelControllerOne.getSparkMax().follow(intakeMotorControllerOne.getSparkMax());
+    //CDSWheelControllerTwo.getSparkMax().follow(intakeMotorControllerOne.getSparkMax(), true);
+    intakeMotorControllerTwo.getSparkMax().follow(intakeMotorControllerOne.getSparkMax());
   }
 
-  public void IntakeSwitch(boolean on){    
-    if (on){
-      double intakeSmartSpeed = SmartDashboard.getNumber("Belt Speed", Constants.CDSBeltSpeed);
-      
-      intakeMotorControllerOne.getSparkMax().set(intakeSmartSpeed);
+  public void toggleIntake(boolean reverse) {
+    if (reverse) {
+      intakeMotorControllerOne.getSparkMax().set(-Constants.intakeMotorSpeed);
+      SmartDashboard.putString("Intake Motor Direction", "Reverse");
+      SmartDashboard.putNumber("Intake Motor Speed", -Constants.intakeMotorSpeed);
+    } else {
+      intakeMotorControllerOne.getSparkMax().set(Constants.intakeMotorSpeed);
+      SmartDashboard.putString("Intake Motor Direction", "Forward");
       SmartDashboard.putNumber("Intake Motor Speed", Constants.intakeMotorSpeed);
+    }
+  }
+
+  public void stopIntake() {
+    intakeMotorControllerOne.getSparkMax().set(0.0);
+    SmartDashboard.putNumber("Intake Motor Speed", 0.0);
+  }
+
+  public boolean getDirection() {
+    // true = inverted, false = forward
+    if (intakeMotorControllerOne.getSparkMax().get() > 0) {
+      return false;
     } else {
-      intakeMotorControllerOne.getSparkMax().set(0);
-      SmartDashboard.putNumber("Intake Motor Speed", 0);
+    return true;
     }
   }
-  
-  public void ForwardIntake(){
-    intakeMotorControllerOne.getSparkMax().setInverted(false);
-    SmartDashboard.putString("Intake Motor Direction", "Forward");
+
+  public boolean[] getBeamBreakStatus() {
+    boolean initialStatus = initialBallSensor.get();
```

Comment:
Suggestion: rename intialStatus to frontStatus, initial implies starting status and that is not really accurate here.